tsconfig.base.json

```diff
@@ -0,0 +1,44 @@
+{
+  "compileOnSave": false,
+  "compilerOptions": {
```

Comment:
Sorry to chim in, but I just noticed that the package isn't `strict` and was thinking that it could be a great opportunity to enable `"strict": true` in order to prevent mistakes like #311 and #488 :)

Reply:
@rafaelss95 This is already a **very** big change, and I don't want to postpone it any further.
I'm planning to enable this in the future. This is just the first phase in several good changes 🙂